updated r-hdf5r recipe for windows

dipterix opened the pull request:
<!--
Thank you for your pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->
Checklist
* [X] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->

<!--
Please add any other relevant info below:
-->

dipterix committed:
updated r-hdf5r recipe for windows

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.


I do have some suggestions for making it better though...


For **recipe**:

 * License is not an SPDX identifier (or a custom LicenseRef) nor an SPDX license expression.

Documentation on acceptable licenses can be found [here]( https://conda-forge.org/docs/maintainer/adding_pkgs.html#spdx-identifiers-and-expressions ).

dipterix committed:
Bump build number as required

dipterix commented:
@conda-forge-admin, please rerender

Commit:
MNT: Re-rendered with conda-build 3.22.0, conda-smithy 3.21.2, and conda-forge-pinning 2022.10.23.10.16.33

dipterix committed:
Fixing license

dipterix committed:
Merge branch 'main' of https://github.com/dipterix/r-hdf5r-feedstock

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.

dipterix commented:
@conda-forge/r , @ocefpaf Please consider this PR. This PR adds support to compile the official CRAN  package `hdf5r` on the Windows system. 

Summarize what's changed:

* Added support to compile `hdf5r` on Windows by adding the native `gcc` compiler to build requirement
* Fixed license & added license file copied from `hdf5r` official repository as suggested by `lint`
* Added `conda-forge/r` to maintainer list (seems like one of recent requirements for new recipes)
* Some CI scripts are auto-generated by the conda-forge bot when I requested re-render (see comments above)

This proposed recipe adds more dependencies compared to the existing version. Some might not be necessary (for example `r-bit`). I think there is space to trim the dependent list. However, I would really appreciate if you could merge this PR first and then make adjustments later. There are quite a few packages waiting for this change ; )

Thanks

ocefpaf commented:
Awesome!
